xonsh/tools.py

```diff
@@ -37,8 +37,10 @@
     string_types = (str, unicode)
     unicode_type = unicode
 
-DEFAULT_ENCODING = sys.getdefaultencoding()
-
+if os.name == 'nt':
```

Comment:
Can you move this a couple of lines down and use the `ON_WINDOWS` variable, please?